### PR TITLE
Add accesscontrol tests

### PR DESCRIPTION
**Pull request by sebrandon1**
* Adding unit tests.
* Removing unnecessary local variables in some funcs.
* Using `StringInSlice` to remove some lookup loops.

### CODE REVIEW

File: test-network-function/accesscontrol/suite.go [hunk 1]
```diff
@@ -189,30 +187,32 @@ func runTestOnPods(env *config.TestEnvironment, testCmd testcases.BaseTestCase,
...
 		lineFields := strings.Split(line, ",")
-		if len(lineFields) != expectedNumFields {
+		//nolint:gomnd
+		if len(lineFields) != 2 {
 			return crNamespaces, fmt.Errorf("failed to parse output line %s", line)
 		}
-		crNamespaces[lineFields[crNameFieldIdx]] = lineFields[namespaceFieldIdx]
+		// 0 = CR Name
+		// 1 = CR Namespace
+		crNamespaces[lineFields[0]] = lineFields[1]
```
Comment (greyerof) [inline test-network-function/accesscontrol/suite.go:215]: I'm a bit confused with this re-hardcoding of slice indexes/lenghts: aren't they magic numbers that should be avoided (as the linter suggests)?

Comment (sebrandon1) [reply test-network-function/accesscontrol/suite.go]: Yes, I agree. I brought the constants back.

File: test-network-function/accesscontrol/suite_test.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	for _, tc := range testCases {
+		crMap, err := parseCrOutput(tc.rawOutput)
+		assert.Equal(t, tc.expectedErr, err)
+		if err == nil {
+			assert.Equal(t, tc.expectedOutput, crMap)
+		}
+	}
+}
+
+func TestGetCrsNamespaces(t *testing.T) {
```
Comment (edcdavid) [inline test-network-function/accesscontrol/suite_test.go:72]: Is it worth keeping this test?

**Review by edcdavid — APPROVED** [review]

File: test-network-function/accesscontrol/suite_test.go (hunk 2 above)
Comment (sebrandon1) [reply test-network-function/accesscontrol/suite_test.go:72]: Why remove it?

**Review by greyerof — APPROVED** [review]
I'm not sure why it is necessary to remove helper variables (podName, podNamespace), as they are harmful and they usually allow to make a lot of statements (like printfs) much shorter.

File: test-network-function/accesscontrol/suite_test.go (hunk 2 above)
Comment (edcdavid) [reply test-network-function/accesscontrol/suite_test.go:72]: no need to remove it, just a suggestion. Already approved

**Comment by sebrandon1** [comment]
I can agree with that.  On the other hand the code doesn't need to allocate new memory (although small and temporary) for doing prints.  If anything, the loop variable `podUnderTest` is too long, and could be shorted to `p`?  Just ideas, maybe for another PR.

File: test-network-function/accesscontrol/suite_test.go (hunk 2 above)
Comment (sebrandon1) [reply test-network-function/accesscontrol/suite_test.go:72]: Sounds good!

**Pull request merged**